x/clp/keeper/provider_distribution.go

```diff
@@ -0,0 +1,117 @@
+package keeper
+
+import (
+	"fmt"
+
+	"github.com/Sifchain/sifnode/x/clp/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+type ProviderDistributionMap map[string]sdk.Uint
+
+func (k Keeper) ProviderDistributionPolicyRun(ctx sdk.Context) {
+	pdMap := k.doProviderDistribution(ctx)
+	for lpAddress, pdRowan := range pdMap {
+		address, err := sdk.AccAddressFromBech32(lpAddress)
+		if err != nil {
+			k.Logger(ctx).Error(fmt.Sprintf("Liquidity provider address %s error %s", lpAddress, err.Error()))
+			continue
+		}
+
+		err = k.transferProviderDistribution(ctx, address, pdRowan)
+		if err != nil {
+			k.Logger(ctx).Error(fmt.Sprintf("Paying out liquidity provider %s error %s", address, err.Error()))
+		}
+	}
+}
+
+func (k Keeper) doProviderDistribution(ctx sdk.Context) ProviderDistributionMap {
+	blockHeight := ctx.BlockHeight()
+	params := k.GetProviderDistributionParams(ctx)
+	if params == nil {
```

Comment:
Is this overly defensive? Can params be nil?

Reply:
`params` can't be nil for now but since `k.GetProviderDistributionParams(ctx)` returns a pointer, it could be nil if somebody ever were to change its implementation. I know, probably unlikely but still. I have mixed feelings about this and I'm sure I'm not consistent with handling of cases like these